File: androidunsplash/src/main/java/com/kc/unsplash/api/HeaderInterceptor.java
package com.kc.unsplash.api;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class HeaderInterceptor implements Interceptor {

    private String clientId;

    public HeaderInterceptor(String clientId) {
        this.clientId = clientId;
    }

    @Override public Response intercept(Interceptor.Chain chain) throws IOException {
        Request request = chain.request();
        request = request.newBuilder()
                .addHeader("Authorization", "Client-ID " + clientId)
                .build();
        return chain.proceed(request);
    }
}

File: androidunsplash/src/main/java/com/kc/unsplash/api/HeaderInterceptor.kt
package com.kc.unsplash.api

import java.io.IOException

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response

class HeaderInterceptor(private val clientId: String, private val token: String?) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        var auth = "Client-ID $clientId"
        if (token != null) {
            auth += ",Bearer $token"
        }
        var request = chain.request()
        request = request.newBuilder()
                .addHeader("Authorization", auth)
                .build()
        return chain.proceed(request)
    }
}